substrate/frame/revive/src/exec.rs

```diff
@@ -574,6 +574,10 @@ pub struct Stack<'a, T: Config, E> {
 /// For each nested contract call or instantiate one frame is created. It holds specific
 /// information for the said call and caches the in-storage `ContractInfo` data structure.
 struct Frame<T: Config> {
+	/// The address who instantiated the call.
+	caller: Origin<T>,
+	/// The address of contract code being executed.
+	callee: H160,
```

Comment:
This is what we have account_id for. Just one line below.

Reply:
account_id is the contract being executed while callee is the address of the code being used. It's the same in case of a call and different on delegate_call. By comparing them I can figure out if we're delegating and I can use the callee for immutables

Reply:
In most of the cases they will be the same. I don't see how this simplifies things.